### DISCUSSION TITLE
Fatrop options

### DISCUSSION THREAD

**Discussion by ajoardar007** [casadi/casadi#3956] [Q&A]
Hello everyone,

I am a graduate-student at the University of Waterloo. 
I have just gotten my optimal control problem's constraint's structured the way FATROP likes it, but now I am looking for ways to pass options, like `constrain violation tol`, `acceptable tol`, etc.

At the moment, I am using Matlab to test it out, so, for IPOPT, the options can be entered as such:
```
% Options
opts = struct;
opts.ipopt.print_level= 0;
opts.print_time= 0;
opts.ipopt.sb='yes';
opts.ipopt.print_level = 6;
opts.ipopt.max_iter = 400;
opts.ipopt.tol = 1e-8;
opts.ipopt.constr_viol_tol = 1e-8;
opts.ipopt.acceptable_tol = 1e-2;
opts.ipopt.acceptable_obj_change_tol = 1e-1;
% Create an NLP solver
prob = struct('f', J, 'x', vertcat(w{:}), 'g', vertcat(G{:}), 'p', vertcat(P{:}));
solver = nlpsol('solver', 'ipopt', prob,opts);
```
But if I replaced `ipopt` with `fatrop`, it doesn't work (I sort of hoped it would because, in the end, [this](https://www.youtube.com/watch?v=X-nS4i8jJyM) introductory-video mentioned that, under the hood, FATROP calls IPOPT to do the solving). Error output:

```
Error using casadi.Function/call
Error in Function::call for 'solver' [FatropInterface] at .../casadi/core/function.cpp:1432:
Error in Function::call for 'solver' [FatropInterface] at .../casadi/core/function.cpp:361:
.../casadi/interfaces/fatrop/fatrop_interface.cpp:570: Fatrop option not supported: acceptable_obj_change_tol
```

Then I tried looking through the [source-file](https://github.com/casadi/casadi/blob/main/casadi/interfaces/fatrop/fatrop_interface.cpp) `fatrop_interface.cpp` on the Casadi-repo, to see if it mentions any list of valid options that I can pass, but all I get is a function-call to `fatrop_ocp_c_option_type()` with the options-argument to check if it is a valid-option or not (and then to throw-out that line above you see).

I am not able to locate the function-definition for `fatrop_ocp_c_option_type`, so i was hoping to learn from others what options are available, and how to pass them.

Also, is there a way to ask the solver to stop and regurgitate the current best-solution after a finite amount of time has passed?

Regards,
Arnab

**Comment by rtapia8484** [comment]
the following seems to list all options: /fatrop/solver/FatropOptions.hpp
